_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -504,6 +504,64 @@ public static function register_endpoints() {
 				'permission_callback' => __CLASS__ . '::view_admin_page_permission_check',
 			)
 		);
+
+		// Update settings from the Jetpack wizard.
+		register_rest_route(
+			'jetpack/v4',
+			'/setup/questionnaire',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => __CLASS__ . '::update_setup_questionnaire',
+				'permission_callback' => __CLASS__ . '::update_settings_permission_check',
+				'args'                => array(
+					'option_values' => array(
+						'required' => true,
+						'type'     => 'object',
+					),
+				),
+			)
+		);
+
+		// Get settings for the Jetpack wizard.
+		register_rest_route(
+			'jetpack/v4',
+			'/setup/questionnaire',
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => __CLASS__ . '::get_setup_questionnaire',
+				'permission_callback' => __CLASS__ . '::update_settings_permission_check',
+			)
+		);
+	}
+
+	/**
+	 * Update the settings selected on the wizard questionnaire
+	 *
+	 * @param WP_REST_Request $request The request.
+	 *
+	 * @return bool true.
+	 */
+	public static function update_setup_questionnaire( $request ) {
+		$option_values = $request['option_values'];
+		update_option( 'setup_questionnaire', $option_values );
+		return true;
+	}
+
+	/**
+	 * Get the settings for the wizard questionnaire
+	 *
+	 * @param WP_REST_Request $request The request.
+	 *
+	 * @return array Questionnaire settings.
+	 */
+	public static function get_setup_questionnaire( $request ) {
+		$options_values = get_option( 'setup_questionnaire' );
```

Comment:
What do you think about storing this option with the rest of the Jetpack options, in the `jetpack_options` option? This allows us to keep things clean on each site, to delete all Jetpack related options quickly when the user uninstalls, and it avoids conflicts with other plugins that may also use generic option names such as `setup_questionnaire`.

You can use `Jetpack_Options::get_option` and `Jetpack_Options::update_option` to do this.

You'll also want to add that new option to the array of options acceptable and synced in that big `jetpack_options` option, here:
https://github.com/Automattic/jetpack/blob/1df89f0d1c38ac1053dea58a66236dd6206d4f77/packages/options/legacy/class-jetpack-options.php#L111